### PR TITLE
openstack: promote externallb job to GA

### PR DESCRIPTION
**Pull request by EmilienM**
- Revert the FailureDomain testing, the implementation will change so we no longer want to test this unsupported interface.
- Add `FEATURE_SET` interface to `openstack-conf-installconfig`
  - This interface will be useful to specify the featureset at the job level, instead of hardcoding it in the steps directly.
- externallb-techpreview: use the new `FEATURE_SET` interface
  - Switch the techpreview jobs for external LB to use this interface now available for the OpenStack jobs.
- openstack: move externallb job to GA
  - The job is TechPreview in 4.13 and becomes GA in 4.14.

### CODE REVIEW

**Comment by EmilienM** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.14-e2e-openstack-externallb periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.13-e2e-openstack-externallb-techpreview

**Comment by EmilienM** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.14-e2e-openstack-externallb periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.13-e2e-openstack-externallb-techpreview

**Comment by EmilienM** [comment]
4.14 is good: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/39921/rehearse-39921-periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.14-e2e-openstack-externallb/1664282408213549056/artifacts/e2e-openstack-externallb/openstack-conf-installconfig/artifacts/install-config.yaml
4.13 not working yet, had to fix the jq command.

**Comment by EmilienM** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.13-e2e-openstack-externallb-techpreview

**Comment by EmilienM** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.13-e2e-openstack-externallb-techpreview

**Comment by pierreprinetti** [comment]
/lgtm
Thanks!

**Comment by EmilienM** [comment]
/pj-rehearse ack
All is good on my side.

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml [hunk 1]
```diff
@@ -2303,21 +2303,21 @@ presubmits:
           secretName: result-aggregator
     trigger: (?m)^/test( | .* )e2e-openstack,?($|\s.*)
   - agent: kubernetes
-    always_run: false
+    always_run: true
```
Comment (sinnykumari) [inline ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml:2306]: any reason to run them on all PRs?

**Comment by sinnykumari** [comment]
LGTM for MCO specific changes
/lgtm

**Comment by sinnykumari** [comment]
Somehow my lgtm triggered both approval and lgtm label. Putting hold in case there is still some review needed. If not please remove the hold
/hold

**Comment by EmilienM** [comment]
This is good to me.
/hold cancel

Thanks for the review!

**Pull request merged**